src/message_conversions.rs

```diff
@@ -0,0 +1,530 @@
+                ..Default::default()
+            };
+
+            let source = UUri {
+                authority_name: ME_AUTHORITY.to_string(), // TODO: Should we set this to anything specific?
+                ue_id: client_id as u32,
+                ue_version_major: 1, // TODO: I don't see a way to get this
+                resource_id: 0,      // set to 0 as this is the resource_id of "me"
+                ..Default::default()
+            };
+
+            // TODO: Not sure where to get this
+            //  Steven said Ivan posted something to a Slack thread; need to check
+            //  Hmm, didn't find this. Asked Steven for help
+            //  He pointed me to something about SOME/IP-SD, but not Request AFAICT
+            let ttl = 1000;
+
+            trace!("Prior to building Request");
+
+            let umsg_res = UMessageBuilder::request(sink, source, ttl)
+                .build_with_payload(payload_bytes, UPayloadFormat::UPAYLOAD_FORMAT_UNSPECIFIED);
+
+            trace!("After building Request");
+
+            let Ok(umsg) = umsg_res else {
+                return Err(UStatus::fail_with_code(
+                    UCode::INTERNAL,
+                    "Unable to build UMessage from vsomeip message",
+                ));
+            };
+
+            let req_id = umsg.attributes.id.as_ref().ok_or_else(|| {
+                UStatus::fail_with_code(
+                    UCode::INVALID_ARGUMENT,
+                    "Missing id for Request message. Would be unable to correlate. Rejected.",
+                )
+            })?;
+            trace!("{} - (req_id, request_id) to store for later correlation in ME_REQUEST_CORRELATION: ({}, {})",
+                UP_CLIENT_VSOMEIP_FN_TAG_CONVERT_VSOMEIP_MSG_TO_UMSG,
+                req_id.to_hyphenated_string(), request_id
+            );
+            let mut me_request_correlation = ME_REQUEST_CORRELATION.write().await;
+            if me_request_correlation.get(req_id).is_none() {
+                trace!("{} - (req_id, request_id) to store for later correlation in ME_REQUEST_CORRELATION: ({}, {})",
+                    UP_CLIENT_VSOMEIP_FN_TAG_CONVERT_VSOMEIP_MSG_TO_UMSG,
+                    req_id.to_hyphenated_string(), request_id
+                );
+                me_request_correlation.insert(req_id.clone(), request_id);
+            } else {
+                return Err(UStatus::fail_with_code(
+                    UCode::ALREADY_EXISTS,
+                    format!(
+                        "Already exists same MT_REQUEST with id: {req_id}, therefore rejecting"
+                    ),
+                ));
+            }
+
+            Ok(umsg)
+        }
+        message_type_e::MT_NOTIFICATION => {
+            trace!("MT_NOTIFICATION type");
+
+            // TODO: Talk with @StevenHartley. It seems like vsomeip notify doesn't let us set the
+            //  interface_version... going to set this manually to 1 for now
+            let interface_version = 1;
```

Comment:
It seems like vsomeip MT_NOTIFICATION doesn't include the interface_version / ue_version_major so I set it to 1. Any better ideas here? 

Tagging @stevenhartley / @int0x27